buildlib/az-helpers.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash -eE
+
+# The following functions uses Azure logging commands to report test
+# details or errors. If the process is not running in Azure environment,
+# no special output is generated.
+
+# Logging commands documentation: https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands
+
+
+RUNNING_IN_AZURE="yes"
+if [ -z "$AGENT_ID" ]; then
+    RUNNING_IN_AZURE="no"
+fi
+
+# Report error and exit
+function error() {
+    msg=$1
+    azure_log_issue "${msg}"
+    echo "ERROR: ${msg}"
+    exit 1
+}
+
+# Define Azure pipeline variable
+function azure_set_variable() {
+    test "x$RUNNING_IN_AZURE" = "xno" && return
+    name=$1
+    value=$2
+    echo "##vso[task.setvariable variable=${name}]${value}"
+}
+
+# Report an issue to Azure pipeline and stop step execution
+function azure_log_issue() {
+    test "x$RUNNING_IN_AZURE" = "xno" && return
+    msg=$1
+    echo "##vso[task.logissue type=error]${msg}"
+    echo "##vso[task.complete result=Failed;]"
+}
+
+# Get IPv4 address of an interface
+function get_ip() {
```

Comment:
why only ipv4? can we test ipv6 as well?

Reply:
We surely can, but that would be different cases. When we add them this function should be updated or a new one added.

Reply:
ok so let's keep it in a TODO list.
we had cases where there were bugs with ipv6 so it's important we add it please.